src/transformers/models/iris/modeling_iris.py

```diff
@@ -0,0 +1,1846 @@
+
+        return (x + h_,) + (attention_weights,)
+
+
+class IrisLPIPS(nn.Module):
+    # Learned perceptual metric
+    def __init__(self, use_dropout: bool = True):
+        super().__init__()
+        self.scaling_layer = IrisScalingLayer()
+        self.chns = [64, 128, 256, 512, 512]  # vgg16 features
+        self.net = IrisVgg16(requires_grad=False)
+        self.lin0 = IrisNetLinLayer(self.chns[0], use_dropout=use_dropout)
+        self.lin1 = IrisNetLinLayer(self.chns[1], use_dropout=use_dropout)
+        self.lin2 = IrisNetLinLayer(self.chns[2], use_dropout=use_dropout)
+        self.lin3 = IrisNetLinLayer(self.chns[3], use_dropout=use_dropout)
+        self.lin4 = IrisNetLinLayer(self.chns[4], use_dropout=use_dropout)
+        for param in self.parameters():
+            param.requires_grad = False
+
+    def normalize_tensor(self, x: torch.Tensor, eps: float = 1e-10) -> torch.Tensor:
+        norm_factor = torch.sqrt(torch.sum(x**2, dim=1, keepdim=True))
+        return x / (norm_factor + eps)
+
+    def spatial_average(self, x: torch.Tensor, keepdim: bool = True) -> torch.Tensor:
+        return x.mean([2, 3], keepdim=keepdim)
+
+    def forward(self, input: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
+        in0_input, in1_input = (self.scaling_layer(input), self.scaling_layer(target))
+        outs0, outs1 = self.net(in0_input), self.net(in1_input)
+        feats0, feats1, diffs = {}, {}, {}
+        lins = [self.lin0, self.lin1, self.lin2, self.lin3, self.lin4]
+        for kk in range(len(self.chns)):
+            feats0[kk], feats1[kk] = self.normalize_tensor(outs0[kk]), self.normalize_tensor(outs1[kk])
+            diffs[kk] = (feats0[kk] - feats1[kk]) ** 2
+
+        res = [self.spatial_average(lins[kk].model(diffs[kk]), keepdim=True) for kk in range(len(self.chns))]
+        val = res[0]
+        for i in range(1, len(self.chns)):
+            val += res[i]
+        return val
+
+
+class IrisScalingLayer(nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+        self.register_buffer("shift", torch.Tensor([-0.030, -0.088, -0.188])[None, :, None, None])
+        self.register_buffer("scale", torch.Tensor([0.458, 0.448, 0.450])[None, :, None, None])
+
+    def forward(self, inp: torch.Tensor) -> torch.Tensor:
+        return (inp - self.shift) / self.scale
+
+
+class IrisNetLinLayer(nn.Module):
+    """A single linear layer which does a 1x1 conv"""
+
+    def __init__(self, chn_in: int, chn_out: int = 1, use_dropout: bool = False) -> None:
+        super().__init__()
+        layers = [nn.Dropout()] if (use_dropout) else []
+        layers += [nn.Conv2d(chn_in, chn_out, 1, stride=1, padding=0, bias=False)]
+        self.model = nn.Sequential(*layers)
+
+
+class IrisVgg16(nn.Module):
+    def __init__(self, requires_grad: bool = False) -> None:
+        super().__init__()
+        vgg_pretrained_features = models.vgg16(weights=models.VGG16_Weights.IMAGENET1K_V1).features
```

Comment:
If you need vgg16, then this should be implemented as layers in this modeling file or loaded as a backbone